src/candy-machine/candy-machine.service.ts

```diff
@@ -446,6 +451,36 @@ export class CandyMachineService {
     return rawTransaction.toString('base64');
   }
 
+  async findByAddress(address: string) {
+    const candyMachine = await this.prisma.candyMachine.findUnique({
+      where: { address },
+    });
+
+    if (!candyMachine) {
+      throw new NotFoundException(
+        `Candy Machine with address ${address} does not exist`,
+      );
+    }
+
+    return candyMachine;
+  }
+
+  async findReceipts(query: CandyMachineReceiptParams) {
+    const receipts = await this.prisma.candyMachineReceipt.findMany({
+      skip: query.skip,
+      take: query.take,
+      where: { candyMachineAddress: query.candyMachineAddress },
+    });
+
+    if (!receipts || !receipts.length) {
+      throw new NotFoundException(
+        `Receipts with Candy Machine address ${query.candyMachineAddress} does not exist`,
+      );
+    }
```

Comment:
It's fine if there are no receipts, it just means we will return an empty array `[ ]` rather than an error.
Frontend will check if the array is empty and if it is add a custom handler "No items minted yet"